packages/@aws-cdk/aws-lambda-event-sources/lib/stream.ts

```diff
@@ -89,8 +89,8 @@ export interface StreamEventSourceProps {
 /**
  * Use an stream as an event source for AWS Lambda.
  */
-export abstract class StreamEventSource implements lambda.IEventSource {
-  protected constructor(protected readonly props: StreamEventSourceProps) {
+export abstract class StreamEventSource<T extends StreamEventSourceProps> implements lambda.IEventSource {
```

Comment:
The build system for cdk projects is the `tsc` compiler. We use `jsii` that enables us to ship CDK in the various languages we support.

jsii does not yet support generics. While this will work for typescript/javascript, it will not work for other languages. Unfortunately, this will have to go back to not using generics.

Reply:
Unfortunately I have no clue how to do this. When I remove the generics the code will fail with:

```
[2021-01-18T16:10:22.198] [ERROR] jsii/compiler - Type model errors prevented the JSII assembly from being created
error JSII5004: "@aws-cdk/aws-lambda-event-sources.ManagedKafkaEventSource#props" changes the property type to "@aws-cdk/aws-lambda-event-sources.ManagedKafkaEventSourceProps" when overriding @aws-cdk/aws-lambda-event-sources.StreamEventSource. Change it to "@aws-cdk/aws-lambda-event-sources.StreamEventSourceProps"
error JSII5004: "@aws-cdk/aws-lambda-event-sources.SelfManagedKafkaEventSource#props" changes the property type to "@aws-cdk/aws-lambda-event-sources.SelfManagedKafkaEventSourceProps" when overriding @aws-cdk/aws-lambda-event-sources.StreamEventSource. Change it to "@aws-cdk/aws-lambda-event-sources.StreamEventSourceProps"
Error: /Users/jan/Code/aws-cdk/node_modules/jsii/bin/jsii --silence-warnings=reserved-word exited with error code 1
```

Please show me how to do this @nija-at.

Reply:
I think I found a solution